processing/src/main/java/org/apache/druid/query/aggregation/any/NumericAnyVectorAggregator.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.any;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.query.aggregation.VectorAggregator;
+import org.apache.druid.segment.vector.VectorValueSelector;
+
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+
+public abstract class NumericAnyVectorAggregator implements VectorAggregator
+{
+  // Rightmost bit for is null check (0 for is null and 1 for not null)
+  // Second rightmost bit for is found check (0 for not found and 1 for found)
+  @VisibleForTesting
+  static final byte BYTE_FLAG_FOUND_MASK = 0x02;
+  private static final byte BYTE_FLAG_NULL_MASK = 0x01;
+  protected static final int FOUND_VALUE_OFFSET = Byte.BYTES;
+
+  private final boolean replaceWithDefault = NullHandling.replaceWithDefault();
+  protected final VectorValueSelector vectorValueSelector;
+
+  public NumericAnyVectorAggregator(VectorValueSelector vectorValueSelector)
+  {
+    this.vectorValueSelector = vectorValueSelector;
+  }
+
+  /**
+   * Initialize the buffer value given the initial offset position within the byte buffer for initialization
+   */
+  abstract void initValue(ByteBuffer buf, int position);
+
+  /**
+   * Place the primitive value in the buffer given the initial offset position within the byte buffer
+   * at which the current aggregate value is stored.
+   */
+  abstract void putValue(ByteBuffer buf, int position, int row);
```

Comment:
nit: how about `putNonNullRow`?

Reply:
Refactored to `boolean putAnyValueFromRow(ByteBuffer buf, int position, int startRow, int endRow)`